source/http2_stream_manager.c

```diff
@@ -0,0 +1,801 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#if _MSC_VER
+#    pragma warning(disable : 4204) /* non-constant aggregate initializer */
+#endif
+
+#include <aws/common/array_list.h>
+#include <aws/common/device_random.h>
+#include <aws/common/hash_table.h>
+#include <aws/common/logging.h>
+#include <aws/common/ref_count.h>
+#include <aws/http/connection.h>
+#include <aws/http/connection_manager.h>
+#include <aws/http/request_response.h>
+#include <aws/io/channel.h>
+
+#include <aws/http/http2_stream_manager.h>
+#include <aws/http/private/http2_stream_manager_impl.h>
+#include <aws/http/private/request_response_impl.h>
+
+#include <inttypes.h>
+
+#if _MSC_VER
+#    pragma warning(disable : 4204) /* non-constant aggregate initializer */
+#endif
+
+/* Apple toolchains such as xcode and swiftpm define the DEBUG symbol. undef it here so we can actually use the token */
+#undef DEBUG
+
+#define STREAM_MANAGER_LOGF(level, stream_manager, text, ...)                                                          \
+    AWS_LOGF_##level(AWS_LS_HTTP2_STREAM_MANAGER, "id=%p: " text, (void *)(stream_manager), __VA_ARGS__)
+#define STREAM_MANAGER_LOG(level, stream_manager, text) STREAM_MANAGER_LOGF(level, stream_manager, "%s", text)
+
+static void s_stream_manager_start_destroy(struct aws_http2_stream_manager *stream_manager);
+static void s_aws_http2_stream_manager_execute_transaction(struct aws_http2_stream_management_transaction *work);
+
+static struct aws_h2_sm_pending_stream_acquisition *s_new_pending_stream_acquisition(
+    struct aws_allocator *allocator,
+    const struct aws_http_make_request_options *options,
+    aws_http2_stream_manager_on_stream_acquired_fn *callback,
+    void *user_data) {
+    struct aws_h2_sm_pending_stream_acquisition *pending_acquisition =
+        aws_mem_calloc(allocator, 1, sizeof(struct aws_h2_sm_pending_stream_acquisition));
+
+    /* Copy the options and keep the underlying message alive */
+    pending_acquisition->options = *options;
+    pending_acquisition->request = options->request;
+    aws_http_message_acquire(pending_acquisition->request);
+    pending_acquisition->callback = callback;
+    pending_acquisition->user_data = user_data;
+    pending_acquisition->allocator = allocator;
+    return pending_acquisition;
+}
+
+static void s_pending_stream_acquisition_destroy(struct aws_h2_sm_pending_stream_acquisition *pending_acquisition) {
+    if (pending_acquisition == NULL) {
+        return;
+    }
+    if (pending_acquisition->request) {
+        aws_http_message_release(pending_acquisition->request);
+    }
+    aws_mem_release(pending_acquisition->allocator, pending_acquisition);
+    return;
+}
+
+/**
+ * Main actions:
+ * - Acquire stream from user
```

Comment:
trivial: where does this comment belong? apparently not to the function that comes after it